components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -251,17 +226,49 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+// handleSubscriptionDeletion deletes the subscription and removes the finalizer
+func (r *Reconciler) handleSubscriptionDeletion(ctx context.Context, subscription *eventingv1alpha1.Subscription, log *zap.SugaredLogger) error {
+	if utils.ContainsString(subscription.ObjectMeta.Finalizers, Finalizer) {
+		if err := r.Backend.DeleteSubscription(subscription); err != nil {
+			log.Errorw("delete subscription failed", "error", err)
+			// if failed to delete the external dependency here, return with error
+			// so that it can be retried
+			return err
+		}
+
+		// remove our finalizer from the list and update it.
+		subscription.ObjectMeta.Finalizers = utils.RemoveString(subscription.ObjectMeta.Finalizers, Finalizer)
+		if err := r.Client.Update(ctx, subscription); err != nil {
+			events.Warn(r.recorder, subscription, events.ReasonUpdateFailed, "Update Subscription failed %s", subscription.Name)
+			log.Errorw("remove finalizer from subscription failed", "error", err)
+			return err
+		}
+		log.Debug("remove finalizer from subscription succeeded")
+	}
+	return nil
+}
+
+func (r *Reconciler) addFinalizerToSubscription(subscription *eventingv1alpha1.Subscription, log *zap.SugaredLogger) error {
```

Comment:
nit: lets add some docu to this method, it does more than the name is saying (k8s update).

Reply:
done